services/mailer.js

```diff
@@ -17,18 +17,42 @@ if (smtpRequiredProps.some(prop => !process.env[prop])) {
 }
 
 // load all the templates as strings
-const templateStrings = {};
-fs.readdir(path.join(__dirname, 'email'), (err, files) => {
-  if (err) {
-    throw err;
-  }
+const templates = {};
+
+// load the temlates per request during development
+if (process.env.NODE_ENV !== 'production') {
+  templates.render = (name, format = 'txt', context) => new Promise((resolve, reject) => {
+    fs.readFile(path.join(__dirname, 'email', [name, format, 'ejs'].join('.')), (err, file) => {
+      if (err) {
+        return reject(err);
+      }
 
-  files.forEach(file => {
-    fs.readFile(path.join(__dirname, 'email', file), 'utf8', (err, data) => {
-      templateStrings[file] = _.template(data);
+      return resolve(_.template(file)(context));
     });
   });
-});
+} else {
+  templates.data = fs.readdirSync(path.join(__dirname, 'email')).reduce((data, filename) => {
```

Comment:
@jde I made this suggestion instead of the previous version as I felt that it matched the way that the existing express template loading worked.